airbyte_cdk/sources/declarative/manifest_declarative_source.py

```diff
@@ -139,6 +144,35 @@ def __init__(
         # apply additional post-processing to the manifest
         self._post_process_manifest()
 
+        self._config: Mapping[str, Any]
+        self._spec_component: Optional[Spec] = None
+        spec = self._source_config.get("spec")
+        if spec:
+            if "type" not in spec:
+                spec["type"] = "Spec"
+            self._spec_component = self._constructor.create_component(SpecModel, spec, dict())
+            mutable_config = dict(config) if config else {}
```

Comment:
I think we should move this functionality into it's own private method just to organize the code a little. I envision it to roughly look like:
```
def _migrate_config(self, config: Mapping[str, Any]) -> Mapping[str, Any]:
  # convert config into mutable mapping
  # migrate config
  # write config back to file system
  # transform config
  # return config

self._config = self._migrate_config(self._config)
```

something like that. Just so we keep some of the gory logic in its own method instead of the constructor